configs/rt_slam_plugins.yaml

```diff
@@ -4,10 +4,10 @@ plugin_group:
   - name: Kimera-VIO/
     path:
       git_repo: https://github.com/ILLIXR/Kimera-VIO
-      version: 6ea99d33fc2435a8c514049416dcb2f8a0e31692
-  #- name: OpenVINS
-  #  path:
-  #    git_repo: https://github.com/ILLIXR/open_vins.git
-  #    version: c6a4f8835851d21f87add38260e73df63c6354a1
+      version: 3f69ea19e063b0b399cba005067b875bd7373deb
```

Comment:
```suggestion
      version: 648ac892543701abd78a6f4bc56a6b6001f8413f
```

Reply:
I suggest we use a better strategy for managing these commit references.

ILLIXR/ILLIXR (what I'll call "main repo") needs to refer to ILLIXR/audio_pipeline and friends (what I'll call "subrepos"). Currently, we do this by pinning an exact version (commit hash). If the subrepo updates, the main repo does not automatically get that change. This is good for long-term stability, bad for rapid development. If you do want those changes, one has to do these small edits. However, these tiny changes are easily overlooked (as they were here) and can be easily wrong.

Referring to "master" of the subrepos is on the other end of the spectrum. It's great for rapid development and bad for stability. We know that branches of main repo can fall months out-of-date. If the main repo references master of the subrepo, the subrepos will have to support a wide range of main repo versions. If a subrepo updates master, it could break old branches of the main repo, which get the update automatically.

The in-between solution I propose is managing these by semver. Whenever master in a subrepo updates, its maintainer may choose to move the tag x.y to point to the new commit or create a new tag, x.(y+1) or (x+1).0. This way, we explicitly say when we are breaking backwards compatibility (when the minor or major version changes), and non-breaking changes can be pulled in for free, enabling rapid development without compromising deterministic behavior of old branches.

Reply:
![image](https://user-images.githubusercontent.com/928570/107311335-c8492a00-6a53-11eb-9405-1f5b81732be1.png)

Reply:
I agree that we should move to a tag-based system like you suggest. @mhuzai Should the versioning system we use match the milestone release versions to avoid confusion? In that case, we would start from a version `2.X` tag?

Reply:
Well, the components are independent of each other, so they should be able to bump their versions independently. We can tag the last standalone commit 1.0, the current latest commit 2.0, and then bump versions in each subrepo as required.